bindings/swift/Sources/OpenDAL/Operator.swift

```diff
@@ -0,0 +1,93 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import COpenDAL
+
+public enum OperatorError: Error {
+    case failedToBuild
+    case operationError(UInt32)
+}
+
+/// A type used to access almost all OpenDAL APIs.
+public class Operator<S: Service> {
+    var nativeOp: opendal_operator_ptr
+    
+    deinit {
+        withUnsafePointer(to: &nativeOp) { nativeOpPtr in
+            opendal_operator_free(nativeOpPtr)
+        }
+    }
+    
+    /// Creates an operator with the given options.
+    ///
+    /// - Parameter options: The option map for creating the operator.
+    /// - Throws: `OperatorError` value that indicates an error if failed.
+    public init(options: [String : String]) throws {
+        var nativeOptions = opendal_operator_options_new()
+        nativeOp = withUnsafeMutablePointer(to: &nativeOptions) { nativeOptionsPtr in
+            defer {
+                opendal_operator_options_free(nativeOptionsPtr)
+            }
+            
+            for option in options {
+                opendal_operator_options_set(nativeOptionsPtr, option.key, option.value)
+            }
+            
+            return opendal_operator_new(S.scheme, nativeOptionsPtr)
+        }
+        
+        guard nativeOp.ptr != nil else {
+            throw OperatorError.failedToBuild
+        }
+    }
+    
+    /// Blockingly write the data to a given path.
+    ///
+    /// - Parameter data: The content to be written.
+    /// - Parameter path: The destination path for writing the data.
+    /// - Throws: `OperatorError` value that indicates an error if failed to write.
+    public func blockingWrite<D: COWData>(_ data: D, to path: String) throws {
+        let code = data.withBorrowedPointer { dataPointer in
+            // FIXME: The C binding has a bug that assumes any buffer are with
```

Comment:
Hi @Xuanwo, just for emphasizing it, `opendal_operator_blocking_write` currently doesn't copy the passed buffer, which will make the stored data invalid when the original buffer is freed. Do you think we need another issue to track it?

Reply:
That ok. opendal will copy the data byself, it's ok to free the data after this function returned.

Reply:
Nope, I tried but unfortunately it didn’t work. The data read back has the same address as the original one, which has already been freed.

https://github.com/apache/incubator-opendal/blob/6ecb07c3c3634be5a4550f407aebc94d54f89232/bindings/c/src/types.rs#LL141C15-L141C15
As the code shows, `Bytes::from_static` won’t copy the buffer, and the pointer is stored as-is in the KV storage.

Reply:
cc @Ji-Xinyou, it's must be a bug that using `Bytes::from_static` to do convert.

Reply:
https://github.com/apache/incubator-opendal/pull/2473 cc @unixzii @Xuanwo